app/src/main/java/org/sopt/daangnmarket_android/ui/view/write/WriteFragment.kt

```diff
@@ -62,6 +58,8 @@ class WriteFragment : Fragment() {
             if (binding.etTitle.text.isNullOrBlank() || binding.etPrice.text.isNullOrBlank() || binding.etContent.text.isNullOrBlank() || binding.viewmodel?.selectedImageList?.value == null) {
                 Toast.makeText(requireContext(), "채워지지 않은 부분이 있습니다", Toast.LENGTH_SHORT).show()
             } else {
+                // 사실 이게 맞는지도 모르겠..긴한데요..ㅠㅠ 이미지 init 값이 뭔지 모르겠어서.. 주석처리해두었습니다ㅜㅜ
+                // writeViewModel.multipart(title = String(), price = String(), content = String(), image = List<Bitmap>(0,0))
                 startActivity(Intent(requireContext(), MainActivity::class.java))
```

Comment:
이 부분(else문)이 실행되는 조건은 이렇습니다.

> 제목도 입력되어 있고, 가격도 입력되어 있고, 내용도 작성되어 있으며, 이미지도 선택되었을 때 실행

즉, 필요한 값이 모두 채워져 있어야 실행되는 블록입니다. 필요한 값이 다 채워졌다는 것은 서버에 MultiPart Post 요청을 보내도 된다는 의미겠죠? 이 위치에서 ViewModel에 있는 유리님이 작성하신 multipart 함수를 호출하면 됩니다.

```suggestion
                writeViewModel.multipart(
                    requireNotNull(writeViewModel.writeTitle.value),
                    requireNotNull(writeViewModel.writePrice.value),
                    requireNotNull(writeViewModel.writeContent.value),
                    requireNotNull(writeViewModel.selectedImageList.value).map { requireNotNull(it.first.image) }
                )
```

requireNotNull() 이라는 메서드는 인자로 주어진 값이 null이 아닐 때에만 그 값을 그대로 리턴해주는 메서드입니다. 만일 writeViewModel.writeTitle.value가 null이라면 IllegalArgumentException이라는 예외를 throw하는데, 쉽게 생각하면 null이 아닐 때만 문제없이 잘 동작하고 null이라면 뭔가 문제가 있다! 라고 알려주는 메서드라고 생각하면 될 것 같습니다.

사실 이미지 부분은 제가 했어야 할 일이지만 그래도 잘 하셨네요. 👍 

추가로 생각해야 할 점은, 서버와 통신 후 성공했을 때 다시 리스트 화면으로 복귀해야 한다는 점입니다. 그래서 그 부분을 고민해보면 어떨까 싶네요! 힌트를 약간 드리자면, 서버가 201이라는 status를 reponse로 보내면 그 때 현재 액티비티(WriteActivity)를 종료시키는 것입니다. :-)

Reply:
아하 requireNotNull()을 사용하면 되는군요!.. 👍🏻